posts/2017-03-15-faunadb-serverless-authentication.md

```diff
@@ -0,0 +1,46 @@
+
+In the spirit of the Serverless model, the authorizer supplies application content functions with a FaunaDB connection secret that corresponds to the currently logged-in user. This way, there is no possibility of application level bugs impacting data integrity and security.
+
+There is no limit to the data security patterns you can model in FaunaDB. See the tutorials for social graph examples, or follow this space for a multi-user TodoMVC example. For now, the content service just looks up the current user in the database.
+
+These instructions for launching the Serverless Authentication Boilerplate with FaunaDB are based on the `serverless-authentication-boilerplate` README.
+
+## Installing Serverless Authentication
+
+The boilerplate ships with code for a few different identity backends. These steps walk you through installing the service and running it with FaunaDB. The FaunaDB example also integrates with the `test-token` example content service. So once you have it running you can [look at that code](https://github.com/laardee/serverless-authentication-boilerplate/blob/37e4006870c708fa3ef8b64d451a13e2ed93e6f3/test-token/handler.js#L20) to see how your application would use the database.
+
+0. If you haven't yet, `npm install -g serverless` and make sure the standard AWS environment variables are set.
+1. Run `serverless install --url https://github.com/laardee/serverless-authentication-boilerplate`, or clone or download the repository.
+2. Rename `authentication/example.env.yml` to `authentication/env.yml` and set environmental variables. Delete the `CacheTable` entry to avoid provisioning DynamoDB tables you won't be using.
+3. Sign up instantly and [create a database in the FaunaDB dashboard](https://fauna.com/serverless-cloud-sign-up).
+4. Configure `FAUNADB_SECRET` in `authentication/env.yml` with a [server secret](https://fauna.com/documentation#authentication) for your database.
+5. Uncomment `return faunaUser.saveUser(profile);` from `authentication/lib/storage/usersStorage.js`.
+6. Change the last line of `authentication/lib/storage/cacheStorage.js` to `exports = module.exports = faunaCache;`
+7. Change directory to `authentication` and run `npm install`.
+8. Run `serverless deploy` in the authentication folder to deploy authentication service to AWS. Notice the arn of the `authorize` function.
+9. Run `serverless invoke -f schema` to create your FaunaDB schema.
+10. (optional) Change directory to `test-token` and insert the arn of the authorize function to `authorizer/arn` in `serverless.yml`. Then run `serverless deploy` to deploy test-token service.
+
```

Comment:
I think it would be nice to have a link to the whole code project here so that the user can re-check it after he has done all the steps.